lib/max_subarray.py

```diff
@@ -2,11 +2,38 @@
 def max_sub_array(nums):
     """ Returns the max subarray of the given list of numbers.
         Returns 0 if  nums is None or an empty list.
-        Time Complexity: ?
-        Space Complexity: ?
+        Time Complexity: O(n)
+        Space Complexity: O(1)
     """
     if nums == None:
```

Comment:
Small style note: generally, with what we call singletons like `None` (singletons are classes that can only have one instance of a class), we prefer to use `is` to `==` because the latter's functionality can be overwritten with the `__eq__` method and very occasionally you will see unexpected results, particularly with singleton values.
```suggestion
    if nums is None:
```